contrib/config/test/kv_store_xds_delegate_integration_test.cc

```diff
@@ -250,6 +250,25 @@ class KeyValueStoreXdsDelegateIntegrationTest
     // Set up a new Envoy, using the previous Envoy's configuration, and create the test server.
     ConfigHelper helper(version_, *api_,
                         MessageUtil::getJsonStringFromMessageOrError(config_helper_.bootstrap()));
+
+    // Close the connection between Envoy and the xDS FakeUpstreams.
+    closeConnection(sds_connection_);
+    closeConnection(rtds_connection_);
+    // Bring down the xDS FakeUpstreams.
+    getSdsUpstream().reset();
+    getRtdsUpstream().reset();
+
+    // Reset the xDS upstreams to new FakeUpstreams, since we'll re-establish a connection to xDS
+    // after the Envoy server gets re-created and the KV store loaded.
+    fake_upstreams_.pop_back(); // Removes the RTDS upstream.
+    fake_upstreams_.pop_back(); // Removes the SDS upstream.
+    // Adds the new SDS upstream.
+    fake_upstreams_.emplace_back(std::make_unique<FakeUpstream>(
```

Comment:
I can't wait until the test is running, b/c by that time, the Bootstrap config is already created (with the xds cluster endpoints with their port numbers), so I need to know the listening port number of the FakeUpstream prior to Envoy recreation.  However, I don't want the FakeUpstream to start listening on that port until after the Envoy server thinks the FakeUpstream is down and fetches xDS from the KV store instead. After fetching from the KV store, I want to have the FakeUpstream listener tied to the server so it starts listening for incoming connections and gets the config from the xDS FakeUpstream.

This was the only way I could get the tests to do what I wanted.

Reply:
What I think I'm misunderstanding, is why an Envoy (test-server) takedown requires restarting of the upstreams (the xDS servers).
They can keep running, using the same ports as before, and wait for new connections. The restarted Envoy's bootstrap will need to be updated with the previous ports.

Reply:
I thought (maybe misremembering) that the delegate will provide the data in case the xDS server disconnects immediately, but I guess that not responding is an easier way to trigger the use of the delegate.

Reply:
yeah, unfortunately that wouldn't trigger the delegate... the xDS server needs to not respond at Envoy server startup to trigger the delegate